components/Admin/AdminButton.tsx

```diff
@@ -22,52 +26,123 @@ function AdminButton({
+      }
+    })
+  }
+
+  const _excludeFromFeed = () => {
+    const params = {
+      excludeFromHomepage: excludeFromFeedSelectedChoices.includes("homepage"),
+      excludeFromHubs: excludeFromFeedSelectedChoices.includes("hubs"),
+    }
+    
+    excludeFromFeed({
+      unifiedDocumentId,
+      params,
+      onSuccess: () => {
+        setIsOpen(false);
+        setMessage("Excluded from Feed");
+        showMessage({ show: true, error: false });
+      },
+      onError: () => {
+        setIsOpen(false);
+        setMessage("Failed to exclude from feed");
+        showMessage({ show: true, error: true });
+      }
+    })
+  }
+
+  const _makeFeatured = () => {
+    const params = {
+      featureInHomepage: makeFeaturedChoices.includes("homepage"),
+      featureInHubs: makeFeaturedChoices.includes("hubs"),
+    }
+
+    featureDoc({
+      unifiedDocumentId,
+      params,
+      onSuccess: () => {
+        setIsOpen(false);
+        setMessage("Document is featured");
+        showMessage({ show: true, error: false });
+      },
+      onError: () => {
+        setIsOpen(false);
+        setMessage("Failed to feature document");
+        showMessage({ show: true, error: true });
+      }
+    })
+  }
+
+  const _removeFromFeatured = () => {
+    removeDocFromFeatured({
+      unifiedDocumentId,
+      onSuccess: () => {
+        setIsOpen(false);
+        setMessage("Document removed from featured");
+        showMessage({ show: true, error: false });
+      },
+      onError: () => {
+        setIsOpen(false);
+        setMessage("Failed to remove featured document");
+        showMessage({ show: true, error: true });
+      }
+    })
+  }      
 
   const dropdownOpts = [{
     icon: icons.eyeSlash,
     label: "Exclude from Trending",
     value: "exclude",
     isVisible: true,
     onSelect: () => {
-      const params = {
-        excludeFromHomepage: excludeFromFeedSelectedChoices.includes("homepage"),
-        excludeFromHubs: excludeFromFeedSelectedChoices.includes("hubs"),
+      if (!isExcludeSubmenuOpen) {
+        _excludeFromFeed();
       }
-      
-      excludeFromFeed({
-        unifiedDocumentId,
-        params,
-        onSuccess: () => {
-          setMessage("Excluded from Feed");
-          showMessage({ show: true, error: false });
-        },
-        onError: () => {
-          setMessage("Failed to exclude from feed");
-          showMessage({ show: true, error: true });
-        }
-      })
     }
   },{
     icon: icons.eye,
     label: "Include in Trending",
     value: "include",
     isVisible: true,
     onSelect: () => {
-      includeInFeed({
-        unifiedDocumentId,
-        onSuccess: () => {
-          setMessage("Included in Feed");
-          showMessage({ show: true, error: false });
-        },
-        onError: () => {
-          setMessage("Failed to include in feed");
-          showMessage({ show: true, error: true });
-        }
-      })
+      _includeInFeed();
     }
+  },{
```

Comment:
could you run linter on this file?. Formatting looks off.

Reply:
sure. I swear my linter runs but maybe not on tsx files for some reason